pkg/nakadi/nakadi.go

```diff
@@ -0,0 +1,125 @@
+	ConsumerLagSeconds(ctx context.Context, subscriptionID string) (int, error)
+	UnconsumedEvents(ctx context.Context, subscriptionID string) (int, error)
+}
+
+// Client defines client for interfacing with the Nakadi API.
+type Client struct {
+	nakadiEndpoint string
+	http           *http.Client
+}
+
+// NewNakadiClient initializes a new Nakadi Client.
+func NewNakadiClient(nakadiEndpoint string, client *http.Client) *Client {
+	return &Client{
+		nakadiEndpoint: nakadiEndpoint,
+		http:           client,
+	}
+}
+
+func (c *Client) ConsumerLagSeconds(ctx context.Context, subscriptionID string) (int, error) {
+	stats, err := c.stats(ctx, subscriptionID)
+	if err != nil {
+		return 0, err
+	}
+
+	var maxConsumerLagSeconds int
+	for _, partition := range stats.Partitions {
+		if partition.ConsumerLagSeconds > maxConsumerLagSeconds {
+			maxConsumerLagSeconds = partition.ConsumerLagSeconds
+		}
+	}
+
+	return maxConsumerLagSeconds, nil
+}
+
+func (c *Client) UnconsumedEvents(ctx context.Context, subscriptionID string) (int, error) {
+	stats, err := c.stats(ctx, subscriptionID)
+	if err != nil {
+		return 0, err
+	}
+
+	var unconsumedEvents int
+	for _, partition := range stats.Partitions {
+		unconsumedEvents += partition.UnconsumedEvents
+	}
+
+	return unconsumedEvents, nil
+}
+
+type statsResp struct {
+	Items []statsSubscription `json:"items"`
+}
+
+type statsSubscription struct {
+	Partitions []statsPartition `json:"partitions"`
+}
+
+type statsPartition struct {
+	Partiton           string `json:"partition"`
+	State              string `json:"state"`
+	UnconsumedEvents   int    `json:"unconsumed_events"`
+	ConsumerLagSeconds int    `json:"consumer_lag_seconds"`
+	StreamID           string `json:"stream_id"`
+	AssignmentType     string `json:"assignment_type"`
+}
+
+// stats returns the Nakadi stats for a given subscription ID.
+//
+// https://nakadi.io/manual.html#/subscriptions/subscription_id/stats_get
+func (c *Client) stats(ctx context.Context, subscriptionID string) (*statsSubscription, error) {
+	endpoint, err := url.Parse(c.nakadiEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	endpoint.Path = fmt.Sprintf("/subscriptions/%s/stats", subscriptionID)
+
+	q := endpoint.Query()
+	q.Set("show_time_lag", "true")
+	endpoint.RawQuery = q.Encode()
+
+	req, err := http.NewRequest(http.MethodGet, endpoint.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	d, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("[nakadi stats] unexpected response code: %d (%s)", resp.StatusCode, string(d))
+	}
+
+	var result statsResp
+	err = json.Unmarshal(d, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(result.Items) != 1 {
+		return nil, fmt.Errorf("expected 1 subscription, %d returned", len(result.Items))
+	}
+
+	return &result.Items[0], nil
```

Comment:
There is a check above so it looks like only subscriptions for single event type are supported. Maybe this should be explicitly documented.

Reply:
Should be easy enough to extend this to multiple event-type subscriptions by specifying an `event-type` under `matchLabels`. If no `event-type` is specified then it could default to first in list, but this behaviour would need to be documented either way 👍

Reply:
Extended to multiple event types.

It will aggregate over all event types for a subscription by default. Users can optionally add `event-type` to filter for a specific one.